src/Particles/babylon.solidParticleSystem.ts

```diff
@@ -965,31 +963,45 @@
-                            this._vertex.z = this._particle._modelBoundingInfo.boundingBox.vectors[b].z * this._particle.scaling.z;
+                        const modelBoundingInfoVectors = this._particle._modelBoundingInfo.boundingBox.vectorsToRef(Tmp.Vector3);
+                        const v = Tmp.Vector3[8];
+                        const min = Tmp.Vector3[9];
+                        const max = Tmp.Vector3[10];
+                        Vector3.FromFloatsToRef(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE, min);
+                        Vector3.FromFloatsToRef(-Number.MAX_VALUE, -Number.MAX_VALUE, -Number.MAX_VALUE, max);
+
+                        for (var b = 0; b < 8; b++) {
+                            this._vertex.x = modelBoundingInfoVectors[b].x * this._particle.scaling.x;
+                            this._vertex.y = modelBoundingInfoVectors[b].y * this._particle.scaling.y;
+                            this._vertex.z = modelBoundingInfoVectors[b].z * this._particle.scaling.z;
                             this._rotated.x = this._vertex.x * this._particle._rotationMatrix[0] + this._vertex.y * this._particle._rotationMatrix[3] + this._vertex.z * this._particle._rotationMatrix[6];
                             this._rotated.y = this._vertex.x * this._particle._rotationMatrix[1] + this._vertex.y * this._particle._rotationMatrix[4] + this._vertex.z * this._particle._rotationMatrix[7];
                             this._rotated.z = this._vertex.x * this._particle._rotationMatrix[2] + this._vertex.y * this._particle._rotationMatrix[5] + this._vertex.z * this._particle._rotationMatrix[8];
-                            bBox.vectors[b].x = this._particle.position.x + this._cam_axisX.x * this._rotated.x + this._cam_axisY.x * this._rotated.y + this._cam_axisZ.x * this._rotated.z;
-                            bBox.vectors[b].y = this._particle.position.y + this._cam_axisX.y * this._rotated.x + this._cam_axisY.y * this._rotated.y + this._cam_axisZ.y * this._rotated.z;
-                            bBox.vectors[b].z = this._particle.position.z + this._cam_axisX.z * this._rotated.x + this._cam_axisY.z * this._rotated.y + this._cam_axisZ.z * this._rotated.z;
+                            v.x = this._particle.position.x + this._cam_axisX.x * this._rotated.x + this._cam_axisY.x * this._rotated.y + this._cam_axisZ.x * this._rotated.z;
+                            v.y = this._particle.position.y + this._cam_axisX.y * this._rotated.x + this._cam_axisY.y * this._rotated.y + this._cam_axisZ.y * this._rotated.z;
+                            v.z = this._particle.position.z + this._cam_axisX.z * this._rotated.x + this._cam_axisY.z * this._rotated.y + this._cam_axisZ.z * this._rotated.z;
+                            min.minimizeInPlace(v);
+                            max.maximizeInPlace(v);
                         }
-                        bBox._update(this.mesh._worldMatrix);
+
+                        bBox.reConstruct(min, max, this.mesh._worldMatrix);

```

Comment:
updating the code so that the local vectors of the BBox are computing into existing temporary Tmp.Vector3.
As the localBounds are changed, calling `bBox._update` is not enough as it would only update the world stuff, meaning that `minimum`, `maximum`, `extends`, ... becomes inconsistents.
That's why I changed the code to call `reConstruct` instead.

Reply:
Same as before, I'm concerned by perf here